scipy/special/cephes/yn.c

```diff
@@ -98,7 +98,7 @@ double yn(int n, double x)
 	r += 2.0;
 	++k;
     }
-    while (k < n);
+    while (k < n && isfinite(an));
```

Comment:
Since `yn` is a public/exposed interface to second order Bessel function, any reason not to regression test it directly? I believe you've mentioned to me before that testing the special functions comprehensively for correctness may be something that happens outside of SciPy at some point as part of the broader ongoing effort, although this case is pretty easily verifiable with i.e., Wolfram, etc. and fails before/passes after your patch because of the `NaN` behavior you mention.

```diff
--- a/scipy/special/tests/test_basic.py
+++ b/scipy/special/tests/test_basic.py
@@ -4318,3 +4318,11 @@ class TestStirling2:
             denom = stirling2([n], k_entries, exact=True)
             num = denom - stirling2([n], k_entries, exact=False)
             assert np.max(np.abs(num / denom)) < 2e-5
+
+
+def test_yn_gh_20405():
+    # enforce consistency with i.e., Wolfram Alpha
+    # output for BesselY near negative asymptotic
+    # limit
+    actual = cephes.yn(500, 1)
+    assert actual == -np.inf
```

Reply:
Thanks @tylerjereddy. Adding that direct regression test sounds like a good idea.